feat(k8s): kube cluster creation update (#3478)
scaleway/docs-content: containers/kubernetes/how-to/create-cluster.mdx

content:
  h1: How to create a Kubernetes Kapsule cluster
  paragraph: Master the process of creating and managing a Scaleway Kubernetes Kapsule cluster, an advanced managed environment for containerized applications.
tags: kubernetes, kapsule-cluster, kapsule
dates:
  validation: 2024-07-15
  posted: 2021-05-31
categories:
  - kubernetes
---

Scaleway Kubernetes Kapsule provides a managed environment for creating, configuring, and operating a cluster of preconfigured nodes for containerized applications. This service allows you to deploy [Kubernetes](https://kubernetes.io) clusters without the complexity of managing the underlying infrastructure.

Key benefits include:
  * Dynamic scaling of pods based on workload demands.
  * Simplified cluster management via [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/), the Kubernetes command-line tool.

To facilitate cluster administration, Scaleway provides a `.kubeconfig` file, enabling you to manage your cluster locally using `kubectl`. [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) is essential for executing commands against Kubernetes clusters.

<Macro id="requirements" />

> Prerequisites:

Reviewer: ```suggestion
```

> - A Scaleway account with active login to the [console](https://console.scaleway.com)

Reviewer: ```suggestion
- A Scaleway account with access to the [console](https://console.scaleway.com)
```

> - [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) for the relevant organization

Reviewer: ```suggestion
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) for the relevant Organization
```

5. Configure the **Private Network** for the cluster. This ensures secure and isolated network communication. Each cluster is auto-configured with a /22 IP subnet.
   Click **Select Private Network** to:
    * Attach an existing Private Network (VPC) within the same Availability Zone from the drop-down menu.
    * Attach a new Private Network to the cluster.
    <Message type="important">
      The Private Network cannot be detached, and the cluster cannot be moved to another Private Network post-creation.
    </Message>
6. Click **Configure pools** to proceed.

## Pool configuration

This section outlines the settings for your cluster pools.

1. Specify the following for each pool:
    * The **Availability Zone** for the pool's nodes.
    * The **node type** for the pool.
    * Configure **pool options**, including node count and whether to enable autoscaling. Options also include enabling autoheal and linking to a placement group, or you can retain default settings.
        <Message type="tip">
          Unsure about the autoheal feature? [Learn more about autoheal.](/containers/kubernetes/concepts/#autoheal)
        </Message>

>     * Enable full isolation if required.

Reviewer: ```suggestion
    * Enable full isolation, if required.
```

        <Message type="tip">
          Need more information on full isolation? [Learn more about full isolation.](/containers/kubernetes/reference-content/secure-cluster-with-private-network/#what-is-the-difference-between-controlled-isolation-and-full-isolation)
        </Message>
2. Click **Add pool** to integrate the pool into the cluster.

> 3. To add additional pools, click **Expand** and repeat the above steps.

Reviewer: ```suggestion
3. To add more pools, click **Expand** and repeat the steps above.
```

---

meta:
  title: How to create a Kubernetes Kapsule cluster
  description: Learn how to create and manage a Scaleway Kubernetes Kapsule cluster, an advanced managed environment for containerized applications.
content:
  h1: How to create a Kubernetes Kapsule cluster
  paragraph: Master the process of creating and managing a Scaleway Kubernetes Kapsule cluster, an advanced managed environment for containerized applications.
tags: kubernetes, kapsule-cluster, kapsule
dates:

>   validation: 2024-08-08

Author: ```suggestion
  validation: 2024-09-30
```

        </Message>
    * Specify the **Kubernetes version** for your cluster.
4. Enter the **cluster's details**. Provide a name for the cluster. Optionally, you can add a description and tags for better organization.
5. Configure the **Private Network** for the cluster to ensure secure and isolated network communication. Each cluster is auto-configured with a /22 IP subnet. Click **Select Private Network** to:
    * Attach an existing Private Network (VPC) within the same Availability Zone from the drop-down menu.
    * Attach a new Private Network to the cluster.
    <Message type="important">
      The Private Network cannot be detached, and the cluster cannot be moved to another Private Network post-creation.
    </Message>
6. Click **Configure pools** to proceed.

## Pool configuration

This section outlines the settings for your cluster pools. You can configure as many pools for your cluster as you require.

1. Configure the following for each pool:
    * Choose an **Availability Zone** for the pool's nodes.
    * Select the **node type** for the pool.
        <Message type="tip">

>           Need advice in choosing a node type? [Learn more about Kubernetes nodes.](/containers/kubernetes/concepts/#node)

Reviewer 2: ```suggestion
          Need advice on choosing a node type? [Learn more about Kubernetes nodes.](/containers/kubernetes/concepts/#node)
```
